### PR TITLE
Reg1reg1 22 frr upgrade

### PR DESCRIPTION
**Pull request by reg1reg1**
* Added libyang build config for the FRR 7.1 requirement for all VM's
* Added the iptables task in the task list for R1
* Added variable support for libyang
* Added package dependencies for libyang 
* Added persistent iptable support for R1 vm.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@
 
 # IDE files
 .idea
+Computer-Networks/R1/roles/computer-networks/tasks/scratchpad.txt
```
Comment (reg1reg1) [inline .gitignore:5]: Will remove this line in the next commit if it is marked as a defect

**Review by reg1reg1 — COMMENTED** [review]
I feel the code is correct, other reviewers please share your opinion

**Review by treddington — CHANGES_REQUESTED** [review]
dig is in dnstools which is missing

File: .gitignore (hunk 1 above)
Comment (reg1reg1) [reply .gitignore:5]: Dnstools not required as net-tools serves the purpose

**Comment by reg1reg1** [comment]
> dig is in dnstools which is missing

**Pull request merged**